### PR TITLE
feat(medium-pass-1): refactor DetailsViewActionMessageCreator assessment methods to AssessmentActionMessageCreator

### PR DESCRIPTION
**Pull request by lisli1**
#### Details

There are functions in DetailsViewActionMessageCreator that are specific to Assessment. This PR moves them to a new AssessmentActionMessageCreator and updates all references to these functions, in preparation for adding a new QuickAssessActionMessageCreator.

##### Motivation

Feature work

#### Pull request checklist

- [x] Addresses an existing issue: WI 2003322
- [x] Ran `yarn null:autoadd`
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/save-assessment-button.tsx [hunk 1]
```diff
@@ -6,12 +6,14 @@ import { InsightsCommandButton } from 'common/components/controls/insights-comma
 import { UserConfigMessageCreator } from 'common/message-creators/user-config-message-creator';
 import { NamedFC } from 'common/react/named-fc';
 import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
+import { AssessmentActionMessageCreator } from 'DetailsView/actions/assessment-action-message-creator';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
 import styles from 'DetailsView/components/common-dialog-styles.scss';
 import * as React from 'react';
 
 export interface SaveAssessmentButtonDeps {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
```
Comment (madalynrose) [inline src/DetailsView/components/save-assessment-button.tsx:15]: I think we can safely remove `detailsViewActionMessageCreator` here.

File: src/background/actions/assessment-action-creator.ts [hunk 2]
```diff
@@ -151,6 +151,10 @@ export class AssessmentActionCreator {
             Messages.Visualizations.DetailsView.Initialize,
             this.onDetailsViewInitialized,
         );
+        this.interpreter.registerTypeToPayloadCallback(
```
Comment (madalynrose) [inline src/background/actions/assessment-action-creator.ts:154]: Why is `this.onDetailsViewInitialized` triggered by two different message types?

Comment (lisli1) [reply src/background/actions/assessment-action-creator.ts:154]: Thanks for pointing this out! Discussed this with Wahaj and decided to remove the initialize call from assessment as it isn't really needed.

**Review by waabid — APPROVED** [review]
LGTM

**Comment by waabid** [comment]
> This looks awesome, @lisli1! I added a couple of comments where I think we can remove `detailsViewActionMessageCreator` from `deps`. Also, I don't see anywhere we're sending an `Assessment.Initialized` message. What is this message supposed to indicate and what should happen when it is received? It is odd that right now it triggers DetailsView.Initialize behavior.

For context: the initialize message is for making sure we show the dialog for target change (in assessments) when you open assessment from a new details view tab. This started from us mainly looking to make sure that a user would see the continue/start new dialog when they start a new browsing session (but now extends to show the dialog for a new details view tab).

In this case, we should only have one message that is sent on details view (and handle it with both assessment and medium pass actions in an action-creator).

**Pull request merged**